core/src/lib.rs

```diff
@@ -220,14 +186,136 @@ mod tests {
             .entry("birthDate".to_string())
             .or_insert(Value::String("1958-07-17".to_string()));
 
+        return (kv_body, kv_subject);
+    }
+
+    #[test]
+    fn test_create_credential() -> Result<(), String> {
```

Comment:
Guys, Is it possible to create subtests in Rust with closures, how it can be done in GoLang: https://github.com/knox-networks/gomono/pull/570 @mkatychev @LuisOsta @Ryanmtate ? Thus we could separate VC creation test from VC creating proof and from Creating Verifiable presentation. Not it is implemented in one test, but maybe we can do subtests like in example PR above?

Reply:
Yes, this is possible; I built something like this for the integrations test harness: 

https://github.com/monetae/core/blob/develop/integration/src/lib.rs#L45-L53
https://github.com/monetae/core/blob/develop/integration/src/lib.rs#L140-L146
https://github.com/monetae/core/blob/develop/integration/src/issuer.rs#L229-L232
https://github.com/monetae/core/blob/develop/integration/bin/main.rs#L43-L67

I can help configure something similar

Reply:
Implemented without subtests but with separate tests function. I think table-driven tests can be added later to this test suite.